crates/biome_js_analyze/src/lint/nursery/no_magic_numbers.rs

```diff
@@ -0,0 +1,240 @@
+use biome_analyze::{
+    context::RuleContext, declare_lint_rule, Ast, FixKind, Rule, RuleDiagnostic, RuleSource,
+};
+use biome_console::markup;
+use biome_deserialize_macros::Deserializable;
+use biome_js_syntax::{
+    AnyJsExpression, JsAssignmentExpression, JsCallExpression, JsSyntaxKind, JsUnaryOperator,
+    JsVariableDeclarator,
+};
+use biome_rowan::{AstNode, SyntaxNode, TextRange};
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+const MAX_ARRAY_LENGTH: i64 = (2_i64.pow(32)) - 1;
+
+#[derive(
+    Debug, Default, Clone, Serialize, Deserialize, Deserializable, PartialEq, Eq, JsonSchema,
+)]
+pub struct NoMagicNumbersConfig {
+    detect_objects: bool,
+    enforce_const: bool,
+    ignore: Vec<i64>,
+    ignore_array_indexes: bool,
+    ignore_default_values: bool,
+    ignore_class_field_initial_values: bool,
```

Comment:
Each field must be documented. These options will appear in the configuration schema, so we should provide a description for our users.